nixos/doc/manual/development/option-types.section.md

```diff
@@ -160,6 +160,10 @@ merging is handled.
     may result in [precision loss](https://github.com/NixOS/nix/issues/5733).
     :::
 
+`types.hex`
+
+:   A hexadecimal number.
```

Comment:
It seems that there's a lot more to say, and maybe move it into the section with string-like types?

Reply:
Added more info, let me know what you think :D